ingest/validation/validate_metadata.py

```diff
@@ -307,12 +322,29 @@ def cast_float_type(value):
 def cast_string_type(value):
     """Cast string type per convention where Pandas autodetected a number
     """
-    if isinstance(value, numbers.Number):
+    if value_is_nan(value):
+        # nan indicates missing data; by type, nan is a numpy float
+        # so a separate type check is needed for proper handling
+        return value
+    elif isinstance(value, numbers.Number):
         return str(value)
     else:
         return value
 
 
+def regularize_ontology_id(value):
+    """Regularize ontology_ids for storage with underscore format
+    """
+    try:
+        ontology_shortname, term_id = re.split('[_:]', value)
+        value = ontology_shortname + '_' + term_id
+        return value
+    # when ontology_id is malformed and has no separator -> ValueError
+    # when ontology_id value is empty string -> TypeError
+    except (ValueError, TypeError):
+        return value
```

Comment:
wouldn't we want to fail here, instead of returning the value?

Reply:
added comment in 3ad4b5d to explain that we return the bad value so it can be evaluated with the JSON Schema validator so the end user receives an actionable error message within the context of metadata validation